specs/FileTypePolicy.md

```diff
@@ -0,0 +1,222 @@
+File Type Policy API
+===
+
+# Background
+When saving a file with original SaveFilePicker, a security alert might be
+prompted, because the browser applies the [file type policies](https://learn.microsoft.com/en-us/deployedge/microsoft-edge-security-downloads-interruptions#file-types-requiring-a-gesture)
+to protect end users. However, in a WebView2 build App, when end users try
+to save a file with a certain file extension, they usually can trust the
+host App, domain and file extension. So, we provide the App developers the
+File Type Policy API to manage the file type policies dynamically.
+
+We'd appreciate your feedback.
+
+# Description
+
+We proposed the `CoreWebView2.SaveFileSecurityCheckStarting` event. You can register
+this event to get the file path, file extension and URI source information,
```

Comment:
Confused about the URI source: is this e.g. the URI of the jpg or of the page that called `window.showSaveFilePicker`? (I'm assuming the latter)

Reply:
The SourceUri needs to be renamed:
1. Is this the URI of the page that linked to the file download or the URI of the file download?
2. This URI contains only the scheme and hostname not the path or query. It needs to be renamed to indicate that.
Follow up required after answering 1 above.

Reply:
Consider names depending on outcome of (1) above:
 * FileOriginUri
 * DocumentOriginUri

Reply:
That name would make sense if this was the File's URI, I was thinking of that alignment with the other names too. But since it's the _document's_ URI it would be misleading. I suggested "page" in this comment but later we discussed "document" as the more standard name. So `DocumentOriginUri`?

Reply:
The pattern consistency is good. But checking: the FileOriginUri wouldn't be the file's origin URI, right?

Reply:
IIUC there's two URIs: the jpg, and the document whose script called `window.save`. If this property is named `FileOriginUri`, I'm going to assume it's referring to the jpg. But I think that's not what it refers to?

Reply:
Right I agree with Mike although I'm confused about what this property actually is. DocumentOriginUri should mean the HTML document's URI that linked to the file that we are saving. And FileOriginUri should mean the URI of the file that we are saving. Which is this property?

Reply:
I examined the code and we discussed. We're going with DocumentOriginUri. The DOM method that can raise this event just gets a file handle back and its up to the page to write to it. In that scenario we don't know the URI of the file we're saving and there may not even be a URI for the file we're saving. So what we do know is the origin of the document.